### PR TITLE
Add extra HttpTriggerAttribute constructors for OpenAPI extension

### PR DESCRIPTION
**Pull request by justinyoo**
This PR is to allow `HttpTriggerAttribute` to set the `AuthorizationLevel` by reading an environment variable, like `%AuthLevel%`.

By accepting this PR, it will enable Azure Functions OpenAPI extension to dynamically set the Swagger UI and OpenAPI document access level in the isolated worker environment.

Reference: https://github.com/Azure/azure-functions-openapi-extension/issues/22

It follows the existing convention to read the connection string for Azure Service Bus trigger, Cosmos DB trigger, etc.

As a fallback, if invalid environment key is supplied, the `AuthorizationLevel` is set to `Function` as a default.

Example:

```csharp
[Function("MyFunction1")]
public HttpResponseData Run(
    [HttpTrigger("%AuthLevel%", "GET", "POST")] HttpRequestData req,
    FunctionContext context)
{
    ...
}
```

### CODE REVIEW

**Comment by justinyoo** [comment]
Related: #473

File: extensions/Worker.Extensions.Http/HttpTriggerAttribute.cs [hunk 1]
```diff
@@ -67,5 +91,37 @@ public HttpTriggerAttribute(AuthorizationLevel authLevel, params string[] method
...
+
+            if (!trimmed.StartsWith("%"))
+            {
+                return AuthorizationLevel.Function;
+            }
+
+            if (!trimmed.EndsWith("%"))
+            {
+                return AuthorizationLevel.Function;
+            }
```
Comment (SeanFeldman) [inline extensions/Worker.Extensions.Http/HttpTriggerAttribute.cs:116]: These two conditional are a single prerequisite - if it's a parameterized variable, `%` should appear at the beginning and at the end. Collapsing the two checks into a single if statement would keep them together.

```suggestion
            if ((trimmed.StartsWith("%") && trimmed.EndsWith("%")) is false)
            {
                return AuthorizationLevel.Function;
            }
```

Comment (justinyoo) [reply extensions/Worker.Extensions.Http/HttpTriggerAttribute.cs]: Thanks for the comment. It's updated.

**Review by fabiocav — CHANGES_REQUESTED** [review]
@justinyoo it would be good to discuss the changes, as functionality like this is handled by the underlying binding, and not the worker extension. 

This type of logic would need to be added to the Functions HTTP trigger, using the proper name resolver. In addition, we need to ensure other components that require use of the function key are able to identify the proper authorization level required to invoke the function, so that needs to be properly resolved in the metadata that is managed by the host.

Also, did you validate the functionality with a runtime change (changing the value of the environment variable with a compiled payload, rather than changing the value prior to compilation)? You'll notice that, since this doesn't rely on just forwarding the reference to the host, runtime resolution will not happen as expected.

**Comment by justinyoo** [comment]
> @justinyoo it would be good to discuss the changes, as functionality like this is handled by the underlying binding, and not the worker extension.

Yes, it's only applied to the `HttpTrigger` binding, which is defined in the `Microsoft.Azure.Functions.Worker.Extensions.Http` extension. Is that what you meant?

> This type of logic would need to be added to the Functions HTTP trigger, using the proper name resolver. In addition, we need to ensure other components that require use of the function key are able to identify the proper authorization level required to invoke the function, so that needs to be properly resolved in the metadata that is managed by the host.

Good point! Please let me know where to look at the metadata resolution part.

> Also, did you validate the functionality with a runtime change (changing the value of the environment variable with a compiled payload, rather than changing the value prior to compilation)? You'll notice that, since this doesn't rely on just forwarding the reference to the host, runtime resolution will not happen as expected.

You're right. Although the authLevel was read from the environment (I set it `Anonymous`), the actual bounded value was different from it (it still recognised as `Function`) when it was deployed to Azure.

To sum up, changing the `HttpTriggerAttribute` itself is not sufficient. Where should I look for more? It would be great if you let me know the point.

Just note that, to make the OpenAPI extension barely working, this change doesn't need. However, for devs to control the extension dynamically, this PR should be applied, once our discussion is over.

**Comment by fabiocav** [comment]
> Yes, it's only applied to the HttpTrigger binding, which is defined in the Microsoft.Azure.Functions.Worker.Extensions.Http extension. Is that what you meant?

No. What I mean is that behavior changes like this one must be made to the actual HTTP Trigger packaged with `Microsoft.Azure.WebJobs.Extensions.Http` so no change in logic here would be required.

> Good point! Please let me know where to look at the metadata resolution part.

No changes to the metadata resolution process would be required, this binding would continue to be a façade and just forward the metadata to the underlying binding. The changes would need to take place in https://github.com/Azure/azure-webjobs-sdk-extensions

> Just note that, to make the OpenAPI extension barely working, this change doesn't need. However, for devs to control the extension dynamically, this PR should be applied, once our discussion is over.

Makes sense, but once the change is made to the HTTP binding, the functionality of the trigger should work as-is, as the metadata parsing and configuration resolution would ultimately be handled by the trigger itself.

**Comment by justinyoo** [comment]
> > Yes, it's only applied to the HttpTrigger binding, which is defined in the Microsoft.Azure.Functions.Worker.Extensions.Http extension. Is that what you meant?
> 
> No. What I mean is that behavior changes like this one must be made to the actual HTTP Trigger packaged with `Microsoft.Azure.WebJobs.Extensions.Http` so no change in logic here would be required.

This part is unclear to me. Let me rephrase.

Although we use this isolated worker extension, `Microsoft.Azure.Functions.Worker.Extensions.Http`, we still need to make the change on `Microsoft.Azure.WebJobs.Extensions.Http` because this isolated worker extension only forwards metadata to that "existing" binding, `Microsoft.Azure.WebJobs.Extensions.Http`.

Is that correct?

**Comment by justinyoo** [comment]
I'm closing this PR for now.
